CMakeLists.txt

```diff
@@ -43,9 +43,35 @@ IF(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows" AND NOT CMAKE_SYSTEM_VERSION)
     SET(CMAKE_SYSTEM_VERSION "10.0.18362.0" CACHE INTERNAL "" FORCE)
 ENDIF()
 
+#build universally on macOS
+if ( APPLE )
+    set( CMAKE_OSX_ARCHITECTURES "x86_64;arm64" CACHE STRING "" )
+endif()
+
 # Set project name
 PROJECT(OSP-MAGNUM CXX)
 
+# From: https://crascit.com/2016/04/09/using-ccache-with-cmake/
+find_program( CCACHE_PROGRAM ccache )
```

Comment:
personally i don't think ccache specific things should live in the cmakelists.txt

In fact, this change will most likely screw up the ccache config that i (think...? did i? i guess i don't actually remember if i submitted it) setup in the continuous integration runners.

Instead, ccache support should live in a CMakePresets.json file

https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html

Reply:
Oh I didn't know this was a thing, thank you. I just added it since it was in the template and the GdExtension takes a while to compile without it on my machine.